tldr.sh

```diff
@@ -72,57 +72,60 @@ init_term_cmds() {
             white=$(    tput setaf 7 || tput AF 7    )
 
             onblue=$(   tput setab 4 || tput AB 4    )
-            ongrey=$(   tput setab 7 || tput AB 7    )
-        }
+            ongrey=$(   tput setab 7 || tput AB 7    );;
+        esac
     } 2>/dev/null ||:
 
     # osx's termcap doesn't have italics. The below adds support for iTerm2
     # and is harmless on Terminal.app
     [ "$(get_platform)" = "osx" ] && {
-        italic=$(echo -e "\033[3m")
-        eitalic=$(echo -e "\033[23m")
+        italic=$(printf "\033[3m")
+        eitalic=$(printf "\033[23m")
     }
 }
 
 heading() {
     local line="$*"
-    echo "$bold$red${line:2}$reset"
+    printf "\n  $cyan${line#??}$reset"
 }
 
 quotation() {
     local line="$*"
-    echo "$under${line:2}$reset"
+    printf "  $yellow${line#??}$reset\n"
 }
 
 list_item() {
     local line="$*"
-    echo "$line$reset"
+    printf "  $green$line$reset"
 }
 
 code() {
     local line="$*"
     # I'm sure there's a better way to strip the first and last characters.
-    line="${line:1}"
+    line="${line#?}"
     line="${line%\`}"
     # convert {{variable}} to italics
-    line=${line//\{\{/$italic}
-    line=${line//\}\}/$eitalic}
+    # Check if the shell support buil-in substitutions
+    (${a//}) 2>/dev/null && line=${line//\{\{/$reset$white} || { hash sed 2>/dev/null && line="$(printf "$line" | sed "s/{{/$reset$white/g" )"; }
+    (${a//}) 2>/dev/null && line=${line//\}\}/$reset$$boldred} || { hash sed 2>/dev/null && line="$(printf "$line" | sed "s/}}/$reset$bold$red/g" )"; }
 
-    echo "$bold$line$reset"
+    printf "    $bold$red$line$reset\n"
 }
 
 text() {
     local line="$*"
-    echo "$line"
+    printf "  $line$reset\n"
 }
 
 # an idiot-level recognition of tldr's markdown. Needs improvement, or
 # subcontracting out to a markdown -> ANSI formatting command
 display_tldr() {
     # read one line at a time, don't strip whitespace ('IFS='), and process
     # last line even if it doesn't have a newline at the end
-    while IFS= read -r line || [[ -n "$line" ]]; do
-        start=${line:0:1}           # get the first character
+    while read line || [ -n "$line" ]; do
```

Comment:
Now I'm also asking myself this questions. It's not really needed... It's nevertheless a good practice?